### PR TITLE
fsspec 0.9 migration

### LINKED ISSUE #205: Adjust for fsspec asyn changes and remove fsspec pin <=0.8.7
Based on [#581](https://github.com/intake/filesystem_spec/pull/581) on fsspec, there will be a significant change to the fsspec async API.  Until that work is finalized, adlfs will be pinned to fsspec <=0.8.7.

Pin to be removed when the work to fsspec is complete, and adlfs updated to align with changes.

### PR DESCRIPTION
**Pull request by isidentical**
Resolves #205

### CODE REVIEW

**Comment by isidentical** [comment]
@hayesgb apparently there were 2 cases (one in `_rm_file`, and the other one in `_get_file`) that required a `.rstrip(delimiter)` though I couldn't find why they popped up now. Other than that, if I am not missing anything there is no async code that calls another synchronized async function so I just migrated maybe_sync's into sync

File: dev-requirements.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@ azure-storage-blob>=12.5.0,<12.7.0
 azure-identity
 dask[dataframe]
 docker>=4.1.0,<5.0.0
-fsspec>=0.8.2,<=0.8.7
+fsspec>=0.9,<0.10.0
```
Comment (TomAugspurger) [inline dev-requirements.txt:8]: fsspec doesn't use semantic versioning, so the `<0.10` doesn't add any benefit.

```suggestion
fsspec>=0.9
```

File: requirements.txt [hunk 2]
```diff
@@ -3,6 +3,6 @@ azure-core>=1.5.0
 azure-datalake-store>=0.0.46,<0.1
 azure-identity
 azure-storage-blob>=12.5.0,<12.7.0
-fsspec>=0.8.2,<=0.8.7
+fsspec>=0.9.0,<0.10.0
```
Comment (TomAugspurger) [inline requirements.txt:6]:
```suggestion
fsspec>=0.9.0
```

File: setup.py [hunk 3]
```diff
@@ -33,7 +33,7 @@
         "azure-datalake-store>=0.0.46,<0.1",
         "azure-identity",
         "azure-storage-blob>=12.5.0,<12.7.0",
-        "fsspec>=0.8.2,<=0.8.7",
+        "fsspec>=0.9.0,<0.10.0",
```
Comment (TomAugspurger) [inline setup.py:36]:
```suggestion
        "fsspec>=0.9.0",
```

**Review by TomAugspurger — COMMENTED** [review]
Thanks for working on this.

**Comment by isidentical** [comment]
@hayesgb it would be amazing to have a new release (so that we can install newer versions of adlfs with other types of remotes [eg gcsfs] at the same time)

**Comment by hayesgb** [comment]
Done.

**Comment by isidentical** [comment]
Thanks

**Pull request merged**